### PR TITLE
Fix nav bar buttons being hidden during Create New Contact

### LINKED ISSUE #1805: Create New Contact from Show Group Members has hidden nav bar buttons
- [x] I have searched open and closed issues for duplicates

----------------------------------------

### Bug description
When you have a group chat with an unknown number (unknown contact), it opens a `ABUnknownPersonViewController`, from which you can select Create New Contact. This opens a New Contact view controller, which is Apple build and controlled.

### Steps to reproduce
- Be in a group chat with a number for which you do not have a contact (or delete the contact after being in the group)
- Tap on the group name
- Tap List Group Members
- Tap the group member that is not in your contacts (it shows up as a number instead of a name)
- Tap Create New Contact

**Actual result:** 
New Contact screen with hidden nav buttons
**Expected result:** 
New Contact screen should be shown with Title, Cancel button, and Done button

### Screenshots
screen shot 2017-02-23 at 1 12 50 am

### Device info

**Device**: iPhone 7
**iOS version**: 10.2.1
**Signal version:** Z.Y2.7.1

### PR DESCRIPTION
**Pull request by finder39**
### First time contributor checklist
- [x] I have read the [README](https://github.com/WhisperSystems/Signal-iOS/blob/master/README.md) and [CONTRIBUTING](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md) documents
- [x] I have signed the [Contributor Licence Agreement](https://whispersystems.org/cla/)

### Contributor checklist
- [x] I'm following the [code, UI and style conventions](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md#code-conventions)
- [x] My commits are rebased on the latest master branch
- [x] My commits are in nice logical chunks
- [x] My contribution is fully baked and is ready to be merged as is
- [x] I have tested my contribution on these devices:
 * iOS Simulator, iPhone 7, 10.2
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message

- - - - - - - - - -

### Description
Fixes #1805 

`ABUnknownPersonViewController`'s subsequent Create New Contact controller cannot display correctly when `[UINavigationBar appearance].tintColor` or `[UIBarButtonItem appearance].tintColor` is set. By setting these to their default values before pushing the instantiated `ABUnknownPersonViewController`, we allow the appropriate buttons to display.

### Screenshots
screen shot 2017-02-23 at 1 19 20 am

### CODE REVIEW

**Comment by michaelkirk** [comment]
Thanks for the PR

When I tap on a contact, then hit back I see the navbar is still black, but it should change back to white.

screen shot 2017-03-13 at 7 53 13 pm

**Comment by finder39** [comment]
Good catch! I didn't see that.

It seems there isn't a good way to get `ABUnknownPersonViewController` to work 100% here. What we can do is get it to show the buttons but not show the `New Contact` title, which is enough to allow users to be able to navigate. This way, it will not impact the main app's experience.

I've updated the pull request to as such.

It still changes the back arrow to black for a single screen, but then it self corrects. There is no way around that as you cannot subclass `ABUnknownPersonViewController`, and have no access to do anything until `[ShowGroupMembersViewController viewWillAppear:(BOOL)animated[` is called 😕

**Comment by finder39** [comment]
haha the "+1" is because i edited the image to take out the rest of the number for security reasons, it's a full number.

I'm not sure where the group title went, something gets really screwy with this "add contact" piece. It seems you fix one issue and two more take it's place.

File: Signal/src/view controllers/ShowGroupMembersViewController.m [hunk 1]
```diff
@@ -75,6 +75,15 @@ - (void)viewDidLoad {
                                                   untilCancelled:nil];
 }
 
+- (void)viewWillAppear:(BOOL)animated
+{
+    [super viewWillAppear:animated];
+    
+    if (!self.isMovingToParentViewController) {
```
Comment (finder39) [inline Signal/src/view controllers/ShowGroupMembersViewController.m]: "Can we detect when we're back at Signal?" -> this is how we do that

**Comment by Imperiopolis-Signal** [comment]
Hi! Thanks so much for your contribution. We've since switched to use the ContactsUI framework which does not have this problem, so I'm closing this out.

**Pull request closed without merging**